src/rust/engine/src/nodes.rs

```diff
@@ -1122,7 +1114,7 @@ impl Task {
     params: Params,
     entry: Arc<rule_graph::Entry<Rule>>,
     generator: Value,
-  ) -> NodeResult<Value> {
+  ) -> NodeResult<(Value, TypeId)> {
```

Comment:
The aforementioned change from `externs/mod.rs` to now directly return the `TypeId` eagerly.